edx_toggles/toggles/state/internal/report.py

```diff
@@ -209,9 +209,28 @@ def _get_settings_state():
     return settings_dict
 
 
+def _add_setting(settings_dict, setting_value, setting_name):
+    """
+    Recursively process a setting value and add boolean values to settings_dict.
+
+    Args:
+        settings_dict: Dictionary to store the processed settings
+        setting_value: The value of the setting to process
+        setting_name: The name/path of the setting
+    """
+    if isinstance(setting_value, bool):
+        toggle_response = get_or_create_toggle_response(settings_dict, setting_name)
+        toggle_response["is_active"] = setting_value
+    elif isinstance(setting_value, dict):
+        for dict_key, dict_value in setting_value.items():
+            nested_name = setting_dict_name(setting_name, dict_key)
+            _add_setting(settings_dict, dict_value, nested_name)
+
+
 def _add_settings(settings_dict):
     """
     Fill the `settings_dict`: will only include values that are set to true or false.
+    Now supports arbitrarily deep nested dictionaries with recursion.
```

Comment:
1. The first line of a docstring should be a 1-liner summary.
2. I removed the word "Now". No one needs to know that it didn't support this in the past. I try to avoid temporal comments.
```suggestion
    Fill the `settings_dict`: will only include values that are set to true or false.

    Supports arbitrarily deep nested dictionaries with recursion.
```